packages/core/src/render3/instructions/shared.ts

```
@@ -2021,7 +2021,15 @@ export function loadComponentRenderer(
 export function handleError(lView: LView, error: any): void {
   const injector = lView[INJECTOR];
   const errorHandler = injector ? injector.get(ErrorHandler, null) : null;
-  errorHandler && errorHandler.handleError(error);
+  if (errorHandler !== null) {
+    errorHandler.handleError(error);
+    // tslint:disable-next-line:no-console
+  } else if (typeof console === 'object' && console !== null) {
+    // tslint:disable-next-line:no-console
+    (console.error || console.log)(error);
+  } else {
+    throw error;
+  }
```

It looks like this function is called in one place and we already throw there:

https://github.com/angular/angular/blob/00291ae0543c2cc142479ea2580a90c74e5a63f5/packages/core/src/render3/instructions/shared.ts#L1889-L1900

So the main goal of this function is to communicate the error to the `ErrorHandler` if it's present (which it does). I'm wondering if this change is needed?

Quick comment: we may also consider wrapping the `handleError` into `ngDevMode` to tree-shake away this function in prod builds (that should also avoid keeping a reference to the `ErrorHandler` token).

---

I think @AndrewKushnir has a handler on this one.

---

What happens if `errorHandler` is not present? As of right now the exception is eaten, an this is what this is trying to solve.

---

If the `errorHandler` is not present, it looks like the original error would still be re-thrown in the `detectChangesInternal` function.